src/sparseml/pytorch/sparsification/pruning/modifier_pruning_constant.py

```diff
@@ -0,0 +1,148 @@
+    """
+
+    def create_sparsity_masks(
+        self,
+        tensors: List[Tensor],
+        target: Union[float, List[float]],
+        global_sparsity: bool = False,
+    ) -> List[Tensor]:
+        """
+        :param tensors: tensors to generate constant masks for
+        :param target: not used for constant pruning
+        :param global_sparsity: not used for constant pruning
+        :return: list of masks derived from pruned values of each of the given tensors
+        """
+        return [torch.ne(tensor, 0.0).type(tensor.type()) for tensor in tensors]
+
+
+@PyTorchModifierYAML()
+class ConstantPruningModifier(BasePruningModifier, BaseConstantPruningModifier):
+    """
+    Holds the sparsity level and shape for a given parameter(s) constant while training.
+    Useful for transfer learning use cases.
+
+    | Sample yaml:
+    |   !ConstantPruningModifier
+    |       start_epoch: 0.0
+    |       end_epoch: 10.0
+    |       params: ['re:.*weight']
+    |       log_types: __ALL__
+
+    :param start_epoch: The epoch to start the modifier at
+    :param end_epoch: The epoch to end the modifier at
+    :param update_frequency: Ignored for this modifier
+    :param params: A list of full parameter names or regex patterns of names to apply
+        pruning to.  Regex patterns must be specified with the prefix 're:'. __ALL__
+        will match to all parameters. __ALL_PRUNABLE__ will match to all ConvNd
+        and Linear layers' weights
+    :param log_types: The loggers to allow the learning rate to be logged to,
+        default is __ALL__
+    """
+
+    @staticmethod
+    def from_sparse_model(model: Module) -> List[ScheduledModifier]:
+        """
+        Create constant ks modifiers for all prunable params in the given model
+        (conv, linear) that have been artificially sparsified (sparsity > 40%).
+        Useful for transfer learning from a pruned model.
+
+        :param model: the model to create constant ks modifiers for
+        :return: the list of created constant ks modifiers
+        """
+        prunable = get_prunable_layers(model)
+        modifiers = []
+
+        for name, layer in prunable:
+            weight = getattr(layer, "weight")
+            sparsity = tensor_sparsity(weight)
+
+            if sparsity > 0.1:  # set at 10% sparsity to be threshold for intentional
+                modifiers.append(
+                    ConstantPruningModifier(params=["{}.{}".format(name, "weight")])
+                )
+
+        return modifiers
+
+    def __init__(
+        self,
+        params: Union[str, List[str]],
+        start_epoch: float = -1.0,
+        end_epoch: float = -1.0,
+        update_frequency: float = -1.0,
+        log_types: Union[str, List[str]] = ALL_TOKEN,
+    ):
+        super(ConstantPruningModifier, self).__init__(
+            params=params,
+            start_epoch=start_epoch,
+            end_epoch=end_epoch,
+            end_comparator=-1,
+            update_frequency=-1,
```

Comment:
Should this be `update_frequency=update_frequency` since it already defaults to one in `__init__`?

Reply:
good catch, there are no updates, so taking this out for ConstantPruningModifier

Reply:
actually this is needed for inheritance reasons, and is documented as dead functionality above